### PR TITLE
Feature/bookshelves

### PR DESCRIPTION
**Pull request by Figments**
## Description

Adds the ability to create and manage bookshelves. Does NOT include any migration code—that will be handled separately on my local machine, so nothing gets committed.

## Changes

- adjusted layout of My Library page
- added Bookshelves
- added small work cards
- added Bookshelves page on profiles (only for public shelves!)

## Areas Affected
- [ ] Site Navigation
- [ ] Home
- [ ] Sign In/Out
- [ ] Registration
- [ ] Settings

.
- [x] Profile
- [ ] My Stuff (main page)
- [ ] Editor
- [ ] Browse
- [x] Work Card

.
- [ ] Work Page
- [ ] Blog Page
- [ ] Collections
- [ ] Comments
- [ ] Search
- [ ] Other Pages

.
- [ ] Account Authentication
- [ ] Dashboard
- [ ] Mobile

### CODE REVIEW

File: libs/client/my-library/src/lib/components/library-menu/library-sub-menu/library-sub-menu.component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        public library: ContentLibraryRepository,
+        private profiles: PseudonymsQuery,
+    ) {
+        super(menuPackage, contextMenu);
+        this.content = menuPackage.context;
+    }
+
+    handleClick() {
+        // IMPORTANT: tells the menu how to close, and anything passed in here is given to (menuAction)
+    }
```
Comment (SigmasonicX) [inline libs/client/my-library/src/lib/components/library-menu/library-sub-menu/library-sub-menu.component.ts:38]: Is this meant to have anything?

Comment (Figments) [reply libs/client/my-library/src/lib/components/library-menu/library-sub-menu/library-sub-menu.component.ts:38]: no, but it's required for the context menu library

File: libs/client/ui/src/lib/components/tag-badge/tag-badge.component.ts [hunk 2]
```diff
@@ -1,20 +1,127 @@
...
+                this.isTeen = false;
+                this.isMature = false;
+                this.isExplicit = false;
+            }
+        } else {
+            this.isEveryone = false;
+            this.isTeen = false;
+            this.isMature = false;
+            this.isExplicit = false;
+        }
```
Comment (SigmasonicX) [inline libs/client/ui/src/lib/components/tag-badge/tag-badge.component.ts]: This seems inefficient

Comment (Figments) [reply libs/client/ui/src/lib/components/tag-badge/tag-badge.component.ts]: it's to address the fact that ratings require different colors for each one, and a tag's color is applied above wherever the tag label is checked

File: libs/client/ui/src/lib/components/tag-badge/tag-badge.component.ts [hunk 3]
```diff
@@ -1,20 +1,127 @@
...
+                    case Genres.Tragedy:
+                        return 'Tragedy';
+                    case Genres.ActionAdventure:
+                        return 'Action/Adventure';
+                    case Genres.ScienceFiction:
+                        return 'Science Fiction';
+                    case Genres.SliceOfLife:
+                        return 'Slice of Life';
+                    case Genres.SpeculativeFiction:
+                        return 'Speculative Fiction';
```
Comment (SigmasonicX) [inline libs/client/ui/src/lib/components/tag-badge/tag-badge.component.ts]: Those are already the string values of those genre enums, there's no need for this code

File: libs/client/ui/src/lib/components/tag-badge/tag-badge.component.ts [hunk 4]
```diff
@@ -1,20 +1,127 @@
...
+            }
+            case TagKind.Rating: {
+                if (this.rating === ContentRating.Everyone) {
+                    return 'Everyone';
+                } else if (this.rating === ContentRating.Teen) {
+                    return 'Teen';
+                } else if (this.rating === ContentRating.Mature) {
+                    return 'Mature';
+                } else if (this.rating === ContentRating.Explicit) {
+                    return 'Explicit';
```
Comment (SigmasonicX) [inline libs/client/ui/src/lib/components/tag-badge/tag-badge.component.ts]: Same here

**Review by SigmasonicX — APPROVED** [review]
LGTM now

**Pull request merged**